### PR TITLE
Proxy chunking should handle arbitrarily large chunks

### LINKED ISSUE #1257: EOF when running `docker images`
```
vagrant@ubuntu-14:~/src/github.com/weaveworks/service/client$ docker images
EOF
```

```
vagrant@ubuntu-14:~/src/github.com/weaveworks/service/client$ docker logs weaveproxy
INFO: 2015/08/04 12:03:13.780959 weave proxy git-9c22066bdbc6
INFO: 2015/08/04 12:03:13.781665 Command line arguments: -H unix:///var/run/weave.sock
INFO: 2015/08/04 12:03:13.922939 proxy listening on unix:///var/run/weave.sock
INFO: 2015/08/04 12:07:45.162543 GET /v1.19/containers/json
INFO: 2015/08/04 12:08:12.504623 POST /v1.19/containers/create
INFO: 2015/08/04 12:08:12.504988 Creating container with WEAVE_CIDR ""
INFO: 2015/08/04 12:08:12.655506 POST /v1.19/containers/08a5dd4968be6fd9c5b837d6879368e71c5c66130dba30c393458212c09336ff/attach?stderr=1&stdin=1&stdout=1&stream=1
INFO: 2015/08/04 12:08:12.660203 POST /v1.19/containers/08a5dd4968be6fd9c5b837d6879368e71c5c66130dba30c393458212c09336ff/start
INFO: 2015/08/04 12:08:13.152305 Attaching container 08a5dd4968be6fd9c5b837d6879368e71c5c66130dba30c393458212c09336ff with WEAVE_CIDR "" to weave network
INFO: 2015/08/04 12:08:14.105474 POST /v1.19/containers/08a5dd4968be6fd9c5b837d6879368e71c5c66130dba30c393458212c09336ff/resize?h=93&w=179
INFO: 2015/08/04 12:08:44.951522 GET /v1.19/containers/08a5dd4968be6fd9c5b837d6879368e71c5c66130dba30c393458212c09336ff/json
INFO: 2015/08/04 12:08:45.369241 POST /v1.19/build?cgroupparent=&cpuperiod=0&cpuquota=0&cpusetcpus=&cpusetmems=&cpushares=0&dockerfile=Dockerfile&memory=0&memswap=0&rm=1&t=service_client_server
INFO: 2015/08/04 12:08:51.871940 GET /v1.19/images/service_client_server:latest/get
INFO: 2015/08/04 12:11:27.725064 GET /v1.19/images/json
ERRO: 2015/08/04 12:11:30.682555 Error forwarding chunked response body: chunk too long
INFO: 2015/08/04 12:11:38.628391 GET /v1.19/containers/json
INFO: 2015/08/04 12:11:40.947518 GET /v1.19/images/json
ERRO: 2015/08/04 12:11:43.836726 Error forwarding chunked response body: chunk too long
INFO: 2015/08/04 12:11:54.231773 GET /v1.19/images/json
ERRO: 2015/08/04 12:11:57.098078 Error forwarding chunked response body: chunk too long
INFO: 2015/08/04 12:11:57.227207 GET /v1.19/containers/json
INFO: 2015/08/04 12:12:04.738877 GET /v1.19/containers/weaveproxy/json
INFO: 2015/08/04 12:12:04.743759 GET /v1.19/containers/weaveproxy/logs?stderr=1&stdout=1&tail=all
```

### PR DESCRIPTION
**Pull request by paulbellamy**
Fixes #1257.

At the moment I'm unit testing it to accept a 1 megabyte chunk. Have tried it with a failing `docker images -a`, but it requires having loads of images, so I'm loathe to add it into the integration test suite.

### CODE REVIEW

**Comment by tomwilkie** [comment]
How many images do you need to create to make it fail? Can we make lots of
very light weight ones?

**Comment by paulbellamy** [comment]
Good call, actually can fail it using a few images with large labels.

File: proxy/chunked.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+// Err returns the error, if any, that was encountered during iteration.
+func (cr *ChunkedReader) Err() error {
+	if cr.err == io.EOF {
+		return nil
+	}
+	return cr.err
+}
+
+func (cr *ChunkedReader) beginChunk() (n uint64) {
```
Comment (tomwilkie) [inline proxy/chunked.go]: Is this copied straight from the library?

I think the use of named return value is less readable.

File: proxy/chunked.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+		// We always know when EOF is coming.
+		// If the caller asked for a line, there should be a line.
+		if err == io.EOF {
+			err = io.ErrUnexpectedEOF
+		} else if err == bufio.ErrBufferFull {
+			err = ErrLineTooLong
+		}
+		return nil, err
+	}
+	if len(p) >= maxLineLength {
```
Comment (tomwilkie) [inline proxy/chunked.go]: Is this not redundant, as maxLineLength == r's buffer size?

**Pull request merged**